### PR TITLE
IPv6 tests: added new IPv6 static tests

### PR DESCRIPTION
**Pull request by SvetaHBar**
the tests are added:
- Remove IPv6 static test - failed
- Add static ipv6 with min state
- Edit static ipv6 and prefix
- Add ifaces with same static ipv6 in one transaction
- Add ifaces with same static ipv6 to existing

Signed-off-by: Sveta Haas <sveta.haas@gmail.com>

### CODE REVIEW

File: tests/integration/static_ip_address_test.py [hunk 1]
```diff
@@ -257,3 +277,119 @@ def test_add_static_ipv6_with_no_address():
...
+                        {'ip': IPV6_ADDRESS2, 'prefix-length': 30}
+                    ]
+                }
+            }
+        ]
+    }
+
+    netapplier.apply(copy.deepcopy(desired_state))
+
+    assertlib.assert_state(desired_state)
```
Comment (EdDev) [inline tests/integration/static_ip_address_test.py]: This does not confirm that the previous address is not in the current state.

File: tests/integration/static_ip_address_test.py [hunk 2]
```diff
@@ -257,3 +277,119 @@ def test_add_static_ipv6_with_no_address():
...
+
+
+def test_edit_static_ipv6_address_and_prefix(setup_eth1_ipv6):
+    desired_state = {
+        INTERFACES: [
+            {
+                'name': 'eth1',
+                'type': 'ethernet',
+                'state': 'up',
```
Comment (EdDev) [inline tests/integration/static_ip_address_test.py]: - `state` is not required (you depend on the fixture which already handled it).
- ipv6.enabled may also be dropped (not sure) as it is already enabled by the fixture.

File: tests/integration/static_ip_address_test.py [hunk 3]
```diff
@@ -58,6 +58,26 @@ def setup_eth1_ipv4():
     netapplier.apply(desired_state)
 
 
+@pytest.fixture
+def setup_eth1_ipv6():
```
Comment (EdDev) [inline tests/integration/static_ip_address_test.py]: Multiple tests are using the fixture, therefore, it is better add a teardown to remove the address.
On the other hand, we have a general fixture (in conftest) which takes eth1/2 down and up again before each test.

Ahhh that `cleanup_test_interfaces` needs to be changed, it is a bad too wide fixture.

File: tests/integration/static_ip_address_test.py [hunk 4]
```diff
@@ -257,3 +277,119 @@ def test_add_static_ipv6_with_no_address():
...
+                }
+            }
+        ]
+    }
+    netapplier.apply(copy.deepcopy(desired_state))
+
+    assertlib.assert_state(desired_state)
+
+
+@pytest.mark.xfail
```
Comment (EdDev) [inline tests/integration/static_ip_address_test.py]: Better give a reason and make it strict so we will not miss it when we fix the problem.

File: tests/integration/static_ip_address_test.py [hunk 5]
```diff
@@ -257,3 +277,119 @@ def test_add_static_ipv6_with_no_address():
...
+def test_remove_static_ipv6(setup_eth1_ipv6):
```
Comment (EdDev) [inline tests/integration/static_ip_address_test.py]: It is `disable` not `remove`.

File: tests/integration/static_ip_address_test.py (hunk 4 above)
Comment (SvetaHBar) [reply tests/integration/static_ip_address_test.py]: Done

File: tests/integration/static_ip_address_test.py (hunk 5 above)
Comment (SvetaHBar) [reply tests/integration/static_ip_address_test.py]: Done

File: tests/integration/static_ip_address_test.py (hunk 2 above)
Comment (SvetaHBar) [reply tests/integration/static_ip_address_test.py]: Done

File: tests/integration/static_ip_address_test.py (hunk 1 above)
Comment (SvetaHBar) [reply tests/integration/static_ip_address_test.py]: Fixed

**Comment by SvetaHBar** [comment]
ci test please

File: tests/integration/static_ip_address_test.py [hunk 6]
```diff
@@ -257,3 +279,108 @@ def test_add_static_ipv6_with_no_address():
...
+
+
+def test_edit_static_ipv6_address_and_prefix(setup_eth1_ipv6):
+    setup_eth1_ipv6[INTERFACES][0]['ipv6']['address'] = [
+         {'ip': IPV6_ADDRESS2, 'prefix-length': 64}
+     ]
+
+    netapplier.apply(copy.deepcopy(setup_eth1_ipv6))
+
+    assert setup_eth1_ipv6[INTERFACES][0]['ipv6']['address'] == [
```
Comment (EdDev) [inline tests/integration/static_ip_address_test.py]: This is checking the desired state and not the current one.
The current needs to be read using `show`.

Comment (SvetaHBar) [reply tests/integration/static_ip_address_test.py]: Fixed

File: tests/integration/static_ip_address_test.py [hunk 7]
```diff
@@ -255,3 +277,130 @@ def test_add_static_ipv6_with_no_address():
...
+                        {'ip': IPV6_ADDRESS2, 'prefix-length': 24}
+                    ]
+                }
+            }
+        ]
+    }
+
+    netapplier.apply(copy.deepcopy(desired_state))
+    eth1_desired_state = desired_state[INTERFACES][0]
+    print eth1_desired_state
```
Comment (EdDev) [inline tests/integration/static_ip_address_test.py]: Leftovers

Comment (SvetaHBar) [reply tests/integration/static_ip_address_test.py]: Fixed

**Comment by SvetaHBar** [comment]
ci test please

**Comment by SvetaHBar** [comment]
ci test please

**Pull request merged**